tensorflow/compiler/tf2tensorrt/convert/convert_nodes.h

```diff
@@ -190,6 +190,12 @@ class TRT_ShapedWeights {
 
   string DebugString() const;
 
+  template <typename T>
+  void ToVector(std::vector<T>* output) const {
```

Comment:
Sounds good! I was concerned that if we returned the vector it  would cause an extra copy - would that happen in this case?

Reply:
I don't think there would be an extra copy involved if we return vector as the result. Return value optimization would be applied to avoid the copy. See https://abseil.io/tips/11

But, there would be one copy involved from Tensor to std::vector. It's not required in our case as the tensor is expected to be small, but let's also avoid that by using absl::Span instead of std::vector. Otherwise, others may use this convenience method without realizing that it involves a copy.

Reply:
Will absl::Span also be able to modify the content of the tensor? That could be useful.

I think the name `ToVector` implies a copy of the contents - do you think so and if so should we rename if we use absl::Span instead?

Reply:
Yes, Span should be able provide reference to the Tensor elements without copying. It does that by storing pointers to the original TensorBuffer.

Yes, we should rename it to a more suitable name.

Reply:
Great, thanks.
`tensor_.flat<T>().data()` returns a `const T*`. It is okay to use this for the Span with a `const_cast`()

Reply:
do we need mutable absl::Span? can't we use absl::Span<const T>?

Reply:
No we don't